core/src/Repository/Query/WhereCriteria.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Harmony\Core\Repository\Query;
+
+/**
+ * @psalm-immutable
+ */
+class WhereCriteria {
```

Comment:
@orioljp in PHP 8.1 we don't need to use psalm-immutable anymore.
Instead add `readonly` to each property.